linked-list.rb

```diff
@@ -51,25 +51,52 @@ def delete(val)
    end
 
    def display
-       # Traverse through the list till you hit the "nil" at the end
-       current = @head
-       full_list = []
-       while current.next_node != nil
-           full_list += [current.value.to_s]
-           current = current.next_node
-       end
-       full_list += [current.value.to_s]
-       puts full_list.join("->")
-   end
-
-   def include?(key)
+     # Traverse through the list till you hit the "nil" at the end
+     current = @head
+     full_list = []
+     while current.next_node != nil
+        full_list += [current.value.to_s]
+        current = current.next_node
+     end
+     full_list += [current.value.to_s]
+     puts full_list.join("->")
    end
 
-   def size
-   end
-
-   def max
-   end
+  def include?(key)
+    current = @head
+    # breaks out when we reach last instead of check last
+    while current.next_node != nil
+      return true if current.value == key
+      current = current.next_node
+    end
+    return true if current.value == key
+    return false
+  end
+
+  def size
+    current = @head
+    size = 1
+    while current.next_node != nil
+      size += 1
+      current = current.next_node
+    end
+    size
+  end
+
+  def max
+    current = @head
+    max = [current]
+    while current.next_node != nil
+      current = current.next_node
+      if current.value > max[0].value
+        max[0] = current
+      end
+    end
+    max[0] = current if current.value > max[0].value
+    # i decided to return the node itself, and you can
+    # call value on it to see its value
+    max[0]
+  end
```

Comment:
This looks pretty good, but I don't understand why max is an array of a single node. I think it would have made more sense to say `max = current.value` (assuming there is at least one node in the list) and then we wouldn't need the `max[0]` everywhere